agent/doctor/docker_runtime_healthcheck.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//      http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package doctor
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+	"github.com/cihub/seelog"
+)
+
+type dockerRuntimeHealthcheck struct {
+	mu sync.RWMutex
+	// Status is the container health status
+	Status HealthcheckStatus `json:"Status,omitempty"`
+	// Timestamp is the timestamp when container health status changed
+	TimeStamp time.Time `json:"TimeStamp,omitempty"`
+	// LastStatus is the last container health status
+	LastStatus HealthcheckStatus `json:"lastStatus,omitempty"`
+	// LastTimeStamp is the timestamp of last container health status
+	LastTimeStamp time.Time `json:"LastTimeStamp,omitempty"`
+	client        dockerapi.DockerClient
+}
+
+func NewDockerRuntimeHealthcheck(client dockerapi.DockerClient) *dockerRuntimeHealthcheck {
+	return &dockerRuntimeHealthcheck{
+		client:    client,
+		Status:    HealthcheckStatusInitializing,
+		TimeStamp: time.Now(),
+	}
+}
+
+func (dhc *dockerRuntimeHealthcheck) RunCheck() HealthcheckStatus {
+	res := dhc.client.SystemPing(context.TODO(), time.Second*2)
```

Comment:
please make timeout a constant. could you also add a `TODO` to pass in context as an argument to the method?